TYPO3.Flow/Classes/TYPO3/Flow/Http/Request.php

```diff
@@ -373,6 +417,136 @@ public function getContent($asResource = false)
     }
 
     /**
+     * Set a trusted proxy header.
+     *
+     * This is primarily useful for testing. If you want to set trusted headers, use the configuration setting
+     * "TYPO3.Flow.http.trustedProxies.headers" instead.
+     *
+     * @param string $type One of the HEADER_* constants
+     * @param string $name The header name to trust or an empty string to not trust this header
+     * @return void
+     */
+    public static function setTrustedProxyHeader($type, $name)
+    {
+        if (!in_array($type, array(self::HEADER_CLIENT_IP, self::HEADER_HOST, self::HEADER_PORT, self::HEADER_PROTOCOL))) {
+            return;
+        }
+        self::$trustedProxiesSettings['headers'][$type] = strval($name);
+    }
+
+    /**
+     * Set the trusted proxies IP addresses.
+     * If the argument is an empty array, no proxies are trusted.
+     * If the argument is the string '*', all proxies are trusted.
+     *
+     * This is primarily useful for testing. If you want to set trusted headers, use the configuration setting
+     * "TYPO3.Flow.http.trustedProxies.proxies" instead.
+     *
+     * @param array|string $trustedProxies
+     * @return void
+     */
+    public static function setTrustedProxies($trustedProxies)
+    {
+        if (is_string($trustedProxies) && $trustedProxies !== '*' && $trustedProxies !== '') {
+            $trustedProxies = [$trustedProxies];
+        }
+        if (!is_array($trustedProxies) && $trustedProxies !== '*') {
+            $trustedProxies = [];
+        }
+        self::$trustedProxiesSettings['proxies'] = $trustedProxies;
+    }
+
+    /**
+     * Get the values of trusted proxy header.
+     *
+     * @param string $type One of the HEADER_* constants
+     * @return array|null An array of the values for this header type or NULL if this header type should not be trusted
+     */
+    protected function getTrustedProxyHeaderValues($type)
```

Comment:
In all the places this is used the code actually only works with the **first** item of the array.. Wouldn't it make sense to rename this to `getTrustedProxyHeaderValue()` and have that logic encapsulated here?